### PR TITLE
Server and handler code example in english

### PR DESCRIPTION
**Pull request by VielenDanke**
Detailed code example with comments

### CODE REVIEW

File: en/hello.md [hunk 1]
```diff
@@ -118,7 +118,102 @@ func main() {
...
+First step - creating handler
+
 ```go
+package handler
+
+import (
+	"context"
+	"errors"
+	httpsrv "github.com/unistack-org/micro-server-http/v3"
+	pb "github.com/vielendanke/event-service/proto"
```
Comment (vtolstov) [inline en/hello.md:132]: why this import github.com/vielendanke/event-service/proto ?

File: en/hello.md [hunk 2]
```diff
@@ -118,7 +118,102 @@ func main() {
...
+type GithubHandler struct {}
+
+func NewGithubHandler() *GithubHandler {
+	return &GithubHandler{}
+}
+
+func (h *GithubHandler) LookupUser(ctx context.Context, req *pb.LookupUserReq, rsp *pb.LookupUserRsp) error {
+	if req.GetUsername() == "" || req.GetUsername() != "vtolstov" {
+		httpsrv.SetRspCode(ctx, http.StatusBadRequest)
+		return errors.New("name is not correct")
```
Comment (vtolstov) [inline en/hello.md:145]: wrong return type, must be *pb.Error type

File: en/hello.md [hunk 3]
```diff
@@ -118,7 +118,102 @@ func main() {
...
+package main
+
+import (
+	"context"
+	jsoncodec "github.com/unistack-org/micro-codec-json/v3"
+	httpsrv "github.com/unistack-org/micro-server-http/v3"
+	"github.com/unistack-org/micro/v3"
+	"github.com/unistack-org/micro/v3/logger"
+	"github.com/unistack-org/micro/v3/server"
+	"github.com/vielendanke/event-service/internal/app/eventservice/handler"
```
Comment (vtolstov) [inline en/hello.md:165]: bad import, i'm prefer to have this stuff in micro-tests repo

File: en/hello.md [hunk 4]
```diff
@@ -118,7 +118,102 @@ func main() {
...
+	defer cancel()
+
+	/*
+	Options for service, pass the context with cancel
+	 */
+	options := append([]micro.Option{},
+		micro.Context(ctx),
+	)
+	// Create new service
+	srv := micro.NewService(options...)
```
Comment (vtolstov) [inline en/hello.md:180]: why not pass all options with server and client on NewService ?

File: en/hello.md [hunk 5]
```diff
@@ -118,7 +118,102 @@ func main() {
...
+	eh := handler.NewGithubHandler()
+
+	// register handler on the server
+	if err := pb.RegisterGithubHandler(srv.Server("event-service"), eh); err != nil {
+		logger.Error(ctx, err)
+	}
+
+	// run service on port 6060
+	if err := srv.Run(); err != nil {
+		logger.Error(ctx, err)
```
Comment (vtolstov) [inline en/hello.md:208]: logger.Fatal

File: en/hello.md [hunk 6]
```diff
@@ -118,7 +118,102 @@ func main() {
...
+	); err != nil {
+		logger.Error(ctx, err)
+	}
+	
+	// create new github handler
+	eh := handler.NewGithubHandler()
+
+	// register handler on the server
+	if err := pb.RegisterGithubHandler(srv.Server("event-service"), eh); err != nil {
+		logger.Error(ctx, err)
```
Comment (vtolstov) [inline en/hello.md:203]: logger.Fatal as you don't need empty server without proper handler

File: en/hello.md [hunk 7]
```diff
@@ -118,7 +118,102 @@ func main() {
...
+		)),
+	); err != nil {
+		logger.Error(ctx, err)
+	}
+	
+	// create new github handler
+	eh := handler.NewGithubHandler()
+
+	// register handler on the server
+	if err := pb.RegisterGithubHandler(srv.Server("event-service"), eh); err != nil {
```
Comment (vtolstov) [inline en/hello.md:202]: srv.Server() can be used to return single server

File: en/hello.md [hunk 8]
```diff
@@ -118,7 +118,102 @@ func main() {
...
+
+	// create codec
+	c := jsoncodec.NewCodec()
+
+	// Initialization service
+	if err := srv.Init(
+		micro.Servers(httpsrv.NewServer(
+			server.Name("github-service"),       // server name
+			server.Version("1.0"),               // server version
+			server.Address(":6060"),             // server port
```
Comment (vtolstov) [inline en/hello.md:190]: better to use 8080 port for server and 9090 for metrics

**Review by vtolstov — CHANGES_REQUESTED** [review]

File: en/hello.md (hunk 1 above)
Comment (VielenDanke) [reply en/hello.md:132]: My bad. Fixed

File: en/hello.md (hunk 3 above)
Comment (VielenDanke) [reply en/hello.md:165]: Changed to "github.com/unistack-org/micro-tests/client/http/handler" is it correct? Write it there later?

File: en/hello.md (hunk 4 above)
Comment (VielenDanke) [reply en/hello.md:180]: Fixed

File: en/hello.md (hunk 5 above)
Comment (VielenDanke) [reply en/hello.md:208]: Got it. Fixed

File: en/hello.md (hunk 6 above)
Comment (VielenDanke) [reply en/hello.md:203]: Done

File: en/hello.md (hunk 7 above)
Comment (VielenDanke) [reply en/hello.md:202]: Sounds good)

File: en/hello.md (hunk 8 above)
Comment (VielenDanke) [reply en/hello.md:190]: Done

File: en/hello.md (hunk 2 above)
Comment (VielenDanke) [reply en/hello.md:145]: Done

File: en/hello.md (hunk 3 above)
Comment (vtolstov) [reply en/hello.md:165]: yes, write later, but not forget =)

File: en/hello.md (hunk 2 above)
Comment (vtolstov) [reply en/hello.md:145]: how about this?

File: en/hello.md [hunk 9]
```diff
@@ -118,7 +118,114 @@ func main() {
...
+			server.Codec("application/json", c), // content-type and codec for marshal/unmarshal
+		)),
+       micro.Client(mhttp.NewClient(
+          client.Name("github-client"),
+          client.Context(ctx),
+          client.Codec("application/json", jsoncodec.NewCodec()),
+          client.ContentType("application/json"),
+       )),
+	); err != nil {
+		logger.Error(ctx, err)
```
Comment (vtolstov) [inline en/hello.md:207]: logger.Fatal too

File: en/hello.md [hunk 10]
```diff
@@ -118,7 +118,114 @@ func main() {
...
+			server.Name("github-service"),       // server name
+			server.Version("1.0"),               // server version
+			server.Address(":8080"),             // server port
+			server.Context(ctx),                 // context with cancel
+			server.Codec("application/json", c), // content-type and codec for marshal/unmarshal
+		)),
+       micro.Client(mhttp.NewClient(
+          client.Name("github-client"),
+          client.Context(ctx),
+          client.Codec("application/json", jsoncodec.NewCodec()),
```
Comment (vtolstov) [inline en/hello.md]: use c := jsoncodec.NewCodec, or best - dont create var and use client.Codec("application/json", jsoncodec.NewCodec()) in both places

File: en/hello.md [hunk 11]
```diff
@@ -118,7 +118,114 @@ func main() {
...
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	/*
+	Options for service, pass the context with cancel
+	 */
+   options := append([]micro.Option{},
+      micro.Servers(httpsrv.NewServer()),
```
Comment (vtolstov) [inline en/hello.md]: micro.Server(httpsrv.NewServer(
			server.Name("github-service"),       // server name
			server.Version("1.0"),               // server version
			server.Address(":8080"),             // server port
			server.Context(ctx),                 // context with cancel
			server.Codec("application/json", c))

, best - create all options before NewService. service.Init(..Option) is needed only if you want to reconfigure stuff with some conditions

File: en/hello.md [hunk 12]
```diff
@@ -118,7 +118,114 @@ func main() {
...
+```go
+package main
+
+import (
+	"context"
+    mhttp "github.com/unistack-org/micro-client-http/v3"
+	jsoncodec "github.com/unistack-org/micro-codec-json/v3"
+	httpsrv "github.com/unistack-org/micro-server-http/v3"
+	"github.com/unistack-org/micro/v3"
+    "github.com/unistack-org/micro/v3/client"
```
Comment (vtolstov) [inline en/hello.md]: gofmt

File: en/hello.md [hunk 13]
```diff
@@ -118,7 +118,114 @@ func main() {
...
+type GithubHandler struct {}
 
+func NewGithubHandler() *GithubHandler {
+	return &GithubHandler{}
+}
+
+func (h *GithubHandler) LookupUser(ctx context.Context, req *pb.LookupUserReq, rsp *pb.LookupUserRsp) error {
+	if req.GetUsername() == "" || req.GetUsername() != "vtolstov" {
+		httpsrv.SetRspCode(ctx, http.StatusBadRequest)
+        return &pb.Error{Message: "name is not correct"}
```
Comment (vtolstov) [inline en/hello.md]: gofmt

File: en/hello.md [hunk 14]
```diff
@@ -118,7 +118,114 @@ func main() {
...
+First step - creating handler
+
 ```go
+package handler
+
+import (
+	"context"
+	"errors"
+	httpsrv "github.com/unistack-org/micro-server-http/v3"
+	pb "github.com/vielendanke/event-service/proto"
```
Comment (vtolstov) [inline en/hello.md:132]: change import

**Review by vtolstov — CHANGES_REQUESTED** [review]

File: en/hello.md (hunk 14 above)
Comment (VielenDanke) [reply en/hello.md:132]: Done

File: en/hello.md (hunk 13 above)
Comment (VielenDanke) [reply en/hello.md]: Done

File: en/hello.md (hunk 12 above)
Comment (VielenDanke) [reply en/hello.md]: Yeap, my IDE gone crazy :)

File: en/hello.md (hunk 2 above)
Comment (VielenDanke) [reply en/hello.md]: Did this)

File: en/hello.md (hunk 3 above)
Comment (VielenDanke) [reply en/hello.md]: I Will =)

File: en/hello.md (hunk 10 above)
Comment (VielenDanke) [reply en/hello.md]: Did it

File: en/hello.md (hunk 9 above)
Comment (VielenDanke) [reply en/hello.md:207]: Done

File: en/hello.md (hunk 11 above)
Comment (VielenDanke) [reply en/hello.md]: Sounds good, done

File: en/hello.md (hunk 3 above)
Comment (vtolstov) [reply en/hello.md]: why not changed already ?=)

**Review by vtolstov — APPROVED** [review]

**Comment by vtolstov** [comment]
LGTM, merged

**Pull request merged**